DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+import asyncio
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ethtool.ethtool import Ethtool
+from dent_os_testbed.lib.onlp.onlp_system_info import OnlpSystemInfo
+
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_get_traffic_stats,
+    tgen_utils_setup_streams,
+    tgen_utils_start_traffic,
+    tgen_utils_stop_traffic,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect,
+    tgen_utils_get_loss
+)
+
+
+pytestmark = [pytest.mark.suite_functional_l1,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+
+@pytest.mark.parametrize("speed , duplex",
+                         [
+                            # (10, "full"),
+                            # (10, "half"),
+                            # (100, "full"),
+                            # (100, "half")
+                            # (1000, "full"),
+                            # (1000, "half"),
+                            (10000, "full"),
+                         ])
+async def test_l1_forced_speed_10_full_duplex(testbed, speed, duplex):
+    """
+    Test Name: L1: forced speed
+    Test Suite: suite_functional_bridging
+    Test Overview: Verify speed, duplex configuration with packet forwarding
+    Test Procedure:
+    1. Retrieve tgen and dent device info
+    2. Create bridge entity
+    3. Enslave ports to the created bridge entity
+    4. Set up port duplex, speed
+    5. Verify port duplex and speed per setup
+    6. Set up traffic according to duplex mode
+    7. Send traffic to rx_ports
+    8. Verify received traffic on ports is as expected
+    9. Transmitted rate equals received rate (and matches them configuration)
+    """
+
+    # 1. Retrieve tgen and dent device info
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print.error("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    bridge = "br0"
+
+    # Check dut product name. If it supports the speed parameter
+    model = await OnlpSystemInfo.show(input_data=[{device_host_name: [{}]}], parse_output=True)
+    switch = model[0][device_host_name]['parsed_output']['product_name']
+    if "AS5114" not in switch and speed < 1000:
```

Comment:
I suggest adding some external (e.g. YAML) file with models and port types. Please do not hardcode model names in test's code.